pkg/users/users.js

```diff
@@ -206,7 +206,7 @@ async function getLogins() {
 
         const loggedIn = currentLogins.includes(name);
 
-        const date_fields = splitLine.slice(-5);
+        const date_fields = splitLine.slice(4);
```

Comment:
How does the `lastlog` output look like for a broken case? I'm very hesistant to introduce an assumption about the number of initial columns, as they are variable. Check the difference between a local and an SSH login:
```
mpitt            pts/6    ::1                                       Wed Apr 10 07:38:25 +0200 2024
martin           tty1                                               Sat Apr  6 19:20:11 +0200 2024
```
That's the reason why I picked the last 5 filelds to capture the date.

lastlog is really hard to machine read, it's a terrible output format. I wish it had a `--json` mode or something similar.. :cry: It's also not reasonable to read /var/log/lastlog with the cockpit/JS API. An alternative may be to spawn a python helper which reads the file and spits out JSON information, but as the [file format](https://manpagez.com/man/5/lastlog/) has platform/architecture specific record sizes this also seems hard.

So in the end we are stuck with calling `lastlog` :cry: 

Please send the offending lastlog output, we can hopefully refine the heuristics to make it work.

Reply:
OK! So I may have caused some confusion with this - my apologies 😞 

I was testing these two changes on two systems, one ubuntu and one yocto. Fixing the undefined issue hid the formatting issue in Yocto. Our Yocto build uses busybox `date`, which doesn't support the formatting options. We are going to switch that out for coreutils date so we can fix this. I will revert this change.